gamemodes/terrortown/gamemode/shared/hud_elements/tttvoice/pure_skin_voice.lua

```diff
@@ -0,0 +1,168 @@
+
+    function HUDELEMENT:GetDefaults()
+        baseDefaults["basepos"] = {
+            x = 10 * self.scale,
+            y = 10 * self.scale,
+        }
+
+        return baseDefaults
+    end
+
+    function HUDELEMENT:PerformLayout()
+        self.scale = appearance.GetGlobalScale()
+        self.basecolor = self:GetHUDBasecolor()
+        self.padding = padding * self.scale
+
+        BaseClass.PerformLayout(self)
+    end
+
+    function HUDELEMENT:DrawVoiceBar(ply, xPos, yPos, w, h)
+        local color = ply:GetVoiceColor()
+        local data = ply:GetFakeVoiceSpectrum(24)
+
+        local widthBar = (w - h - self.padding) / #data - 1
+        local heightBar = h - 2 * self.padding
+
+        draw.Box(xPos + self.padding, yPos + self.padding, w - self.padding, heightBar, color)
+        self:DrawLines(
+            xPos + self.padding,
+            yPos + self.padding,
+            w - self.padding,
+            heightBar,
+            color.a
+        )
+
+        for i = 1, #data do
+            local yValue = math.floor(data[i] * 0.5 * heightBar - 4 * self.scale)
+
+            draw.Box(
+                xPos + h + 3 + (i - 1) * (widthBar + 1),
+                yPos + 0.5 * h - yValue - 1,
+                widthBar,
+                yValue,
+                Color(255, 255, 255, 35)
+            )
+            if yValue > 1 then
+                draw.Box(
+                    xPos + h + 3 + (i - 1) * (widthBar + 1),
+                    yPos + 0.5 * h - yValue - 2,
+                    widthBar,
+                    1,
+                    Color(255, 255, 255, 140)
+                )
+            end
+
+            draw.Box(
+                xPos + h + 3 + (i - 1) * (widthBar + 1),
+                yPos + 0.5 * h,
+                widthBar,
+                1,
+                Color(255, 255, 255, 175)
+            )
+
+            draw.Box(
+                xPos + h + 3 + (i - 1) * (widthBar + 1),
+                yPos + 0.5 * h + 2,
+                widthBar,
+                yValue,
+                Color(255, 255, 255, 35)
+            )
+            if yValue > 1 then
+                draw.Box(
+                    xPos + h + 3 + (i - 1) * (widthBar + 1),
+                    yPos + 0.5 * h + 2 + yValue,
+                    widthBar,
+                    1,
+                    Color(255, 255, 255, 140)
+                )
+            end
+        end
+
+        draw.FilteredTexture(
+            xPos,
+            yPos,
+            h,
+            h,
+            draw.GetAvatarMaterial(ply:SteamID64(), "medium"),
+            255,
+            COLOR_WHITE
+        )
+        self:DrawLines(xPos, yPos, h, h, 255)
+
+        draw.AdvancedText(
+            ply:Nick(),
+            "PureSkinPopupText",
+            xPos + h + self.padding,
+            yPos + h * 0.5 - 1,
+            util.GetDefaultColor(color),
+            TEXT_ALIGN_LEFT,
+            TEXT_ALIGN_CENTER,
+            true,
+            self.scale
+        )
```

Comment:
What about very long names? Should we add some kind of ellipsis effect for those? (eg ...)

Reply:
would probably good, but as this is done a frame by frame base and has to use the utf8 lib, it would probably be rather heavy. I could cache it to improve that, but it is still a heavy calculation I think

Reply:
Yes that would be good currently i think this would just overflow the box right?

Reply:
yep

Reply:
could we save such an "elliptic nick" on the player object and either use that directly or create a `ply:NickElliptic()` func? we might want to use something like this in the roundendscreen and the scoreboard as well, right? or even other places

Reply:
huh, neat idea. I just started working on a draw.GetLimitedText function, hmm. `ply:NickElliptic()` needs a parameter for the max text length that could then be used for caching

Reply:
I think I resolved this issue in #1498